### DISCUSSION TITLE
How to add SN nodes (peers)

### DISCUSSION THREAD

**Discussion by h-ahmad** [HewlettPackard/swarm-learning#124] [General]
What about the swop profiles?
What about the user, user/app-data folders?
What about the increase in PEERS from 2 to 3 in swci?
What about the certificates and sharing among peers?
Any other change required?

_Originally posted by @h-ahmad in https://github.com/HewlettPackard/swarm-learning/discussions/111#discussioncomment-3258135_

**Comment by IMOKURI** [comment]
I think you should read documents and try what you think first.

If there is a problem, you can ask here according to the issue template.

**Comment by h-ahmad** [comment]
> I think you should read documents and try what you think first.
> 
> If there is a problem, you can ask here according to the issue template.

I have gone through the documentation but could not find the procedure to increae the number of SN nodes. I also searched in issues/closed issues and discussion. 
However, I have tried to add SN node by following changes:
Before: After
1. There were 2 profiles for 2 nodes in swop folder: I have created a third profile with additional index of 2 in each profile.
2. There were user1 and user2 in first node and user3 & user4 in second node: I have make them up to 3 folders (user1, user2, user3), (user4, user5 user6), (user7, user8, user9) for each 3 nodes.
3. There was number of PEERS as 2 in swci init file: I have modified to 3.

Remaining settings are same as given in the example. But the problem is that when starting process, two nodes (including sentinel and randomly second; 2 or 3) works well. It does not simply enrol the task for node 2/3. I will attached the screenshot here once I produce that error.
Thanks for your response and time.

**Comment by RadhakrishnaJ** [comment]
> > I think you should read documents and try what you think first.
> > If there is a problem, you can ask here according to the issue template.
> 
> I have gone through the documentation but could not find the procedure to increae the number of SN nodes. I also searched in issues/closed issues and discussion. However, I have tried to add SN node by following changes: Before: After
> 
> 1. There were 2 profiles for 2 nodes in swop folder: I have created a third profile with additional index of 2 in each profile.
> 2. There were user1 and user2 in first node and user3 & user4 in second node: I have make them up to 3 folders (user1, user2, user3), (user4, user5 user6), (user7, user8, user9) for each 3 nodes.
> 3. There was number of PEERS as 2 in swci init file: I have modified to 3.
> 
> Remaining settings are same as given in the example. But the problem is that when starting process, two nodes (including sentinel and randomly second; 2 or 3) works well. It does not simply enrol the task for node 2/3. I will attached the screenshot here once I produce that error. Thanks for your response and time.

Hi,
Could you please try with 1.1.0, let us know if issue still exists?

**Comment by h-ahmad** [comment]
> > > I think you should read documents and try what you think first.
> > > If there is a problem, you can ask here according to the issue template.
> > 
> > 
> > I have gone through the documentation but could not find the procedure to increae the number of SN nodes. I also searched in issues/closed issues and discussion. However, I have tried to add SN node by following changes: Before: After
> > 
> > 1. There were 2 profiles for 2 nodes in swop folder: I have created a third profile with additional index of 2 in each profile.
> > 2. There were user1 and user2 in first node and user3 & user4 in second node: I have make them up to 3 folders (user1, user2, user3), (user4, user5 user6), (user7, user8, user9) for each 3 nodes.
> > 3. There was number of PEERS as 2 in swci init file: I have modified to 3.
> > 
> > Remaining settings are same as given in the example. But the problem is that when starting process, two nodes (including sentinel and randomly second; 2 or 3) works well. It does not simply enrol the task for node 2/3. I will attached the screenshot here once I produce that error. Thanks for your response and time.
> 
> Hi, Could you please try with 1.1.0, let us know if issue still exists?

The version has been updated to 1.1.0, but I have made many changes suggested by the issues/solutions here. For example, container_id issue in swop. Anyway, I will try to implement and let you know if issue persist. Thanks.
